universal_transfer_operator/src/universal_transfer_operator/data_providers/filesystem/__init__.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from universal_transfer_operator.constants import FileType, TransferMode
+from universal_transfer_operator.data_providers import create_dataprovider
+from universal_transfer_operator.datasets.file.base import File
+from universal_transfer_operator.utils import TransferParameters
+
+
+def resolve_file_path_pattern(
+    file: File,
+    filetype: FileType | None = None,
+    normalize_config: dict | None = None,
+    transfer_params: TransferParameters = None,
+    transfer_mode: TransferMode = TransferMode.NONNATIVE,
+) -> list[File]:
+    """get file objects by resolving path_pattern from local/object stores
+    path_pattern can be
+    1. local location - glob pattern
+    2. s3/gcs location - prefix
+
+    :param file: File dataset object
+    :param filetype: constant to provide an explicit file type
+    :param normalize_config: parameters in dict format of pandas json_normalize() function
+    :param transfer_params: kwargs to be used by method involved in transfer flow.
+    :param transfer_mode: Use transfer_mode TransferMode; native, non-native or thirdparty.
+    """
+    location = create_dataprovider(
+        dataset=file,
+        transfer_params=transfer_params,
+        transfer_mode=transfer_mode,
+    )
+    files = []
+    for path in location.paths:
+        if not path.endswith("/"):
+            file = File(
+                path=path,
+                conn_id=file.conn_id,
+                filetype=filetype,
+                normalize_config=normalize_config,
+            )
+            files.append(file)
+    if len(files) == 0:
+        raise FileNotFoundError(f"File(s) not found for path/pattern '{file.path}'")
```

Comment:
Should this always return an error? Is there no scenario where someone might run against an empty directory (with the idea that sometimes there are files and sometimes there aren't)

Reply:
It is a valid scenario that a user may want to run against an empty directory in the case of incremental updates. If users transfer deltas using a pattern (e.g. `YYYY/MM/DD/HH`), there could be circumstances when the directory is empty.

It may make sense to expose a flag in the operator to raise an exception if the directory is empty, if the user wants this behaviour, but perhaps that could be a separate task/PR.

Reply:
Hmmm ya, I too think this is a valid scenario. But I think in some scenarios user might want the operator to fail in case there are no files. So would it make sense to expose a parameter to the end user - `fail_if_no_files_found` or something else?

Reply:
default `fail_if_source_empty = False`

Reply:
I'll add a separate ticket for the this.

Reply:
Ticket - https://github.com/astronomer/astro-sdk/issues/1816

Reply:
Ticket - https://github.com/astronomer/astro-sdk/issues/1817